presto-parquet/src/main/java/com/facebook/presto/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -287,18 +291,19 @@ private ValuesReader readPageV2(DataPageV2 page)
     {
         repetitionReader = buildLevelRLEReader(columnDescriptor.getMaxRepetitionLevel(), page.getRepetitionLevels());
         definitionReader = buildLevelRLEReader(columnDescriptor.getMaxDefinitionLevel(), page.getDefinitionLevels());
-        return initDataReader(page.getDataEncoding(), page.getSlice().getBytes(), 0, page.getValueCount());
+        return initDataReader(page.getDataEncoding(), ByteBufferInputStream.wrap(ByteBuffer.wrap(page.getSlice().getBytes())), page.getValueCount());
     }
 
     private LevelReader buildLevelRLEReader(int maxLevel, Slice slice)
     {
         if (maxLevel == 0) {
             return new LevelNullReader();
         }
+
         return new LevelRLEReader(new RunLengthBitPackingHybridDecoder(BytesUtils.getWidthFromMaxInt(maxLevel), new ByteArrayInputStream(slice.getBytes())));
     }
 
-    private ValuesReader initDataReader(ParquetEncoding dataEncoding, byte[] bytes, int offset, int valueCount)
+    private ValuesReader initDataReader(ParquetEncoding dataEncoding, ByteBufferInputStream in, int valueCount)
```

Comment:
We don't abbreviate variable names, so instead of `in` please use `inputStream`.